File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/resolver/ClassResolverInterface.java
package com.github.tobi812.sprykerplugin.models.resolver;

import pav.sprykerFileCreator.model.renderer.dto.PhpClassInterface;

public interface ClassResolverInterface {

    PhpClassInterface resolveBundleClass(String classType, String bundleName) throws Exception;

    PhpClassInterface resolveClass(String fullQualifiedName, Boolean findInterface, Boolean findClassBelow);

    PhpClassInterface resolveClassBelow(String fullQualifiedName);

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/resolver/ClassResolverInterface.kt
package com.github.tobi812.sprykerplugin.models.resolver

import com.github.tobi812.sprykerplugin.models.renderer.dto.PhpClassInterface
import java.lang.Exception

interface ClassResolverInterface {
    @Throws(Exception::class)
    fun resolveBundleClass(classType: String, bundleName: String): PhpClassInterface?
    fun resolveClass(fullQualifiedName: String, findClassBelow: Boolean, findInterface: Boolean): PhpClassInterface?
    fun resolveClassBelow(fullQualifiedName: String): PhpClassInterface?
}